services/bugzilla/bugzilla.service.js

```diff
@@ -29,22 +29,41 @@ module.exports = class Bugzilla extends BaseJsonService {
   static get route() {
     return {
       base: 'bugzilla',
-      pattern: ':bugNumber',
+      pattern: ':protocol(http|https)?/:host?/:path*/:bugNumber',
     }
   }
 
   static get examples() {
     return [
       {
         title: 'Bugzilla bug status',
-        namedParams: { bugNumber: '996038' },
+        namedParams: {
+          protocol: 'https',
+          host: 'bugzilla.mozilla.org',
+          bugNumber: '996038',
+        },
         staticPreview: this.render({
           bugNumber: 996038,
           status: 'FIXED',
           resolution: '',
         }),
         documentation,
       },
+      {
+        title: 'Bugzilla bug status (with path)',
+        namedParams: {
+          protocol: 'https',
+          host: 'bugs.eclipse.org',
+          path: 'bugs',
```

Comment:
I wonder if it would be cleaner to put the URL in the query string. It makes for a bit more consistent way of handling these.

Reply:
Do you have an example of a badge using query strings on top of your head?

I mainly based this on Jenkins's badges, which also use paths to target different instances. We may have inconsistent approaches in different places.

Reply:
I think the majority of services are using path params for custom urls/instances, but a couple of the newest/recently refactor services are using a query param for the url. Few examples with query param for url that I remember:

- [BitbucketPullRequest](https://github.com/badges/shields/blob/master/services/bitbucket/bitbucket-pull-request.service.js#L13)
- [Drone](https://github.com/badges/shields/blob/8621fe42d76b809e9ba4ae844920f8ed0373817a/services/drone/drone-build.service.js#L15)
- [Mastodon](https://github.com/badges/shields/blob/8621fe42d76b809e9ba4ae844920f8ed0373817a/services/mastodon/mastodon-follow.service.js#L14)

Reply:
Thanks for those pointers. I'll propose a revised implementation later in the week!

Reply:
Yea, I think we are handling these in different ways. Some services have a host, some have a host and path, and then there's an optional scheme, which makes the approaches less consistent. I like using the query parameter for this as it's the clearest to understand and easiest to implement, though as a negative, it makes for less pretty URLs.

Reply:
Alternative implementation ready for review! 😁